sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -74,31 +79,56 @@ public static SclRootAdapter updateHeader(@NonNull SCL scd, @NonNull HeaderDTO h
         return sclRootAdapter;
     }
 
-
     public static IEDAdapter addIED(SCL scd, String iedName, SCL icd) throws ScdException {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        return sclRootAdapter.addIED(icd,iedName);
+        return sclRootAdapter.addIED(icd, iedName);
     }
 
-    public static Optional<CommunicationAdapter> addSubnetworks(SCL scd, Set<SubNetworkDTO> subNetworks) throws ScdException {
+    public static Optional<CommunicationAdapter> addSubnetworks(SCL scd, Set<SubNetworkDTO> subNetworks, Optional<SCL> icd) throws ScdException {
```

Comment:
why ICD is optional ?

Reply:
Because according to the standard, the Communication tag is not mandatory in the ICD file, and if the Communication tag is present then it must contain the SubNetwork tag which itself must contain the ConnectedAP tag. The Address and PhysConn tags are optional in ConnectedAP.